lib/ansible/modules/cloud/cloudscale/cloudscale_server.py

```diff
@@ -0,0 +1,500 @@
+
+        return self._info
+
+    def _wait_for_state(self, states):
+        start = datetime.now()
+        timeout = self._module.params['api_timeout'] * 2
+        while datetime.now() - start < timedelta(seconds=timeout):
+            server_info = self._get_server_info(refresh=True)
+            if server_info.get('status') in states:
+                return server_info
+            sleep(1)
+        # Timeout succeeded
+        if server_info.get('name') is not None:
+            msg = "Timeout while waiting for a state change on server %s to states %s. " \
+                  "Current state is %s." % (server_info.get('name'), states, server_info.get('status'))
+        else:
+            name_uuid = self._module.params.get('name') or self._module.params.get('uuid')
+            msg = 'Timeout while waiting to find the server %s' % name_uuid
+
+        self._module.fail_json(msg=msg)
+
+    def _start_stop_server(self, server_info, target_state="running", ignore_diff=False):
+        actions = {
+            'stopped': 'stop',
+            'running': 'start',
+        }
+
+        server_status = server_info.get('status')
+        if server_status != target_state:
+            self._result['changed'] = True
+
+            if not ignore_diff:
+                self._result['diff']['before'].update({
+                    'status': server_info.get('status'),
+                })
+                self._result['diff']['after'].update({
+                    'status': target_state,
+                })
+            if not self._module.check_mode:
+                self._post('servers/%s/%s' % (server_info['uuid'], actions[target_state]))
+                server_info = self._wait_for_state((target_state, ))
+
+        return server_info
+
+    def _create_server(self, server_info):
+        self._result['changed'] = True
+        required_params = ('name', 'ssh_keys', 'image', 'flavor')
+        self._module.fail_on_missing_params(required_params)
+        params = self._module.params
+        data = {
+            'name': params['name'],
+            'image': params['image'],
+            'flavor': params['flavor'],
+            'volume_size_gb': params['volume_size_gb'],
+            'bulk_volume_size_gb': params['bulk_volume_size_gb'],
+            'ssh_keys': params['ssh_keys'],
+            'use_public_network': params['use_public_network'],
+            'use_ipv6': params['use_ipv6'],
+            'anti_affinity_with': params['anti_affinity_with'],
+            'user_data': params['user_data'],
+        }
+        self._result['diff']['before'] = self._init_server_container()
+        self._result['diff']['after'] = deepcopy(data)
+        if not self._module.check_mode:
+            self._post('servers', data)
+            server_info = self._wait_for_state(('running', ))
+        return server_info
+
+    def _has_changed(self, server_info, data):
```

Comment:
probably rename this method to `_get_keys_changed` to reflect the actually behaviour